Source/WebCore/html/HTMLButtonElement.cpp

```diff
@@ -342,4 +337,25 @@ bool HTMLButtonElement::isExplicitlySetSubmitButton() const
     return isSubmitButton() && hasAttributeWithoutSynchronization(HTMLNames::typeAttr);
 }
 
+void HTMLButtonElement::computeType(const AtomString& typeAttrValue)
+{
+    Type oldType = m_type;
+    if (equalLettersIgnoringASCIICase(typeAttrValue, "reset"_s))
+        m_type = RESET;
+    else if (equalLettersIgnoringASCIICase(typeAttrValue, "button"_s))
+        m_type = BUTTON;
+    else if (typeAttrValue.isEmpty() && document().settings().invokerAttributesEnabled()) {
```

Comment:
This seems wrong. Shouldn't this check that it's not the string `submit`?

Reply:
It was correct at the time it was written but yes it now needs changing. Will mark as draft until I've resynced the tests and updated to match the spec.